src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int indexOfLastAddedElement = 0;
```

Comment:
0 is default value for int 
```suggestion
    private int size;
```